src/renderers/WebGLRenderer.js

```diff
@@ -1476,34 +1474,15 @@ function WebGLRenderer( parameters ) {
 
 		if ( programChange ) {
 
-			if ( parameters.shaderID ) {
 
-				var shader = ShaderLib[ parameters.shaderID ];
-
-				materialProperties.shader = {
-					name: material.type,
-					uniforms: UniformsUtils.clone( shader.uniforms ),
-					vertexShader: shader.vertexShader,
-					fragmentShader: shader.fragmentShader
-				};
-
-			} else {
-
-				materialProperties.shader = {
-					name: material.type,
-					uniforms: material.uniforms,
-					vertexShader: material.vertexShader,
-					fragmentShader: material.fragmentShader
-				};
-
-			}
-
-			material.onBeforeCompile( materialProperties.shader, _this );
-
-			// Computing code again as onBeforeCompile may have changed the shaders
-			code = programCache.getProgramCode( material, parameters );
+			materialProperties.shader = {
+				name: parameters.name,
+				uniforms: parameters.uniforms,
+				vertexShader: parameters.vertexShader,
+				fragmentShader: parameters.fragmentShader
+			};
 
-			program = programCache.acquireProgram( material, materialProperties.shader, parameters, code );
+			program = programCache.acquireProgram( material, parameters, code );
 
 			materialProperties.program = program;
 			material.program = program;
```

Comment:
We probably should remove this. A material that is used for non-shadowed objects and shadowed objects would results in two different programs and we would be reassigning this property.

Reply:
Initial version of removal of material.program can be found in #15074